radial_build.py

```diff
@@ -0,0 +1,337 @@
+        title.yml
+        """
+        
+        data_dict = self.__dict__
+
+        filename = self.title.replace(' ', "") + '.yml'
+
+        with open(filename, 'w') as file:
+            yaml.safe_dump(data_dict, file, default_flow_style=False,
+                           sort_keys=False)
+
+    def plot_radial_build(self, title="radial_build", colors=None,
+                          max_characters=35, max_thickness=1e6, size=(8, 4),
+                          unit='cm'):
+        """
+        Creates a radial build plot, with layers scaled between a minimum and
+            maximum pixel width to preserve readability
+
+        Arguments:
+            title (string): title for plot and filename to save to
+            colors (list of str): list of matplotlib color strings. 
+                If specific colors are desired for each layer they can be added
+                here
+            max_characters (float): maximum length of a line before wrapping the 
+                text
+            max_thickness (float): maximum thickness of layer to display, useful
+                for reducing the total size of the figure.
+            size (iter of float): figure size, inches. (width, height)
+            unit (str): Unit of thickness values
+        """
+        if colors is None:
+            colors = list(matplotlib.colors.XKCD_COLORS.values())[
+                0:len(self.build)]
+
+        self.title = title
+        self.colors = colors
+        self.max_characters = max_characters
+        self.max_thickness = max_thickness
+        self.size = size
+        self.unit = unit
+
+        char_to_height = 1.15
+        min_line_height = 8
+        min_lines = 2
+        height = char_to_height*max_characters
+
+        # initialize list for lower left corner of each layer rectangle
+        ll = [0, 0]
+        plt.figure(1, figsize=self.size)
+        plt.tight_layout()
+        ax = plt.gca()
+        ax.set_ylim(0, height+1)
+
+        total_thickness = 0
+        for (name, layer), color in zip(self.build.items(), self.colors):
+            if 'thickness' not in layer or ('thickness' in layer and layer['thickness'] != 0):
+                if 'thickness' not in layer:
+                    layer['thickness'] = min_line_height
+                    thickness_str = ''
+                else:
+                    thickness_str = f': {layer["thickness"]} {self.unit}'
+
+                if 'composition' not in layer:
+                    comp_string = ''
+                else:
+                    comp_string = self.build_composition_string(
+                        layer['composition'])
+
+                if 'description' not in layer:
+                    description_str = ''
+                else:
+                    description_str = textwrap.fill(
+                        f'{layer["description"]}',
+                        self.max_characters,
+                        drop_whitespace=False
+                    )
```

Comment:
move `rstrip()` here

Reply:
since either, both, or neither of the `"composition"` and `"description"` keys may be present the `rstrip()` works best in its current location, I think its the simplest, otherwise logic needs to be in place to check what combination of keys were specified.